src/components/SMARTBox/PatientBox.js

```diff
@@ -83,19 +79,12 @@ export default class SMARTBox extends Component {
     this.props.clearCallback();
     if(this.state.request !== "none" ) {
       const request = JSON.parse(this.state.request);
-      if (request.resourceType === "DeviceRequest") {
-        this.updateDeviceRequest(patient, request);
-      } else if (request.resourceType === "ServiceRequest") {
-        this.updateServiceRequest(patient, request);
-      } else if (request.resourceType === "MedicationRequest") {
-        this.updateMedicationRequest(patient, request);
-      } else if (request.resourceType === "MedicationDispense") {
-        this.updateMedicationDispense(patient, request);
-      } 
-        else {
+      if (request.resourceType === "DeviceRequest" || request.resourceType === "ServiceRequest" || request.resourceType === "MedicationRequest" || request.resourceType === "MedicationDispense") {
```

Comment:
Is the check for list of resourceTypes necessary since you always call the same function? Unless you need to stop other types from calling into it that is.

Reply:
Yeah that's there to prevent other types from calling in